src/roswire/common/msg.py

```diff
@@ -55,10 +50,42 @@ class Constant:
         The value of this constant.
     """
 
+    R_STRING_CONSTANT = re.compile("^\s*string\s+(\w+)\s*=\s*(.+)\s*$")
+    R_OTHER_CONSTANT = re.compile("^\s*(\w+)\s+(\w+)\s*=\s*([^\s]+).*$")
+
     typ: str
     name: str
     value: Union[str, int, float]
 
+    @classmethod
+    def from_string(cls, package: str, line: str) -> "Optional[Constant]":
```

Comment:
It looks like `package` isn't used here?

Reply:
Right. I was doing it to make it consistent with the other from_string that is needed in Field. I also wasn't sure if constants can have types in the same way that Fields can, where you need to resolve a name in the package. (Currently, Constant has a todo on processing the type.)

Reply:
I don't believe that the name of the package is necessary to determine the qualified name of the constant type. (I might be wrong for ROS2?) The FIXME rightly complains about the lack of type conversion in the constant value itself (e.g., integer types aren't converted from `str` to `int`), but I suppose that's a separate issue 🙂

Reply:
It looks like constants even in ROS 2 can only have primitive types. So I've removed the package parameter.